rust-connectors/sources/mqtt/src/main.rs

```diff
@@ -84,11 +97,40 @@ fn main() -> Result<(), MqttConnectorError> {
             %mqtt_topic,
             %client_id
         );
+        let port = if let Some(port) = opts.mqtt_port {
+            port
+        } else {
+            if opts.mqtt_tls {
+                8883
+            } else {
+                1883
+            }
+        };
 
         let timeout = std::time::Duration::from_secs(mqtt_timeout_seconds);
 
-        let mut mqttoptions = MqttOptions::new(client_id, mqtt_url, 1883);
+        let mut mqttoptions = MqttOptions::new(client_id, mqtt_url, port);
+        if let (Some(username), Some(password)) = (opts.mqtt_username, opts.mqtt_password) {
+            mqttoptions.set_credentials(username, password);
+        }
         mqttoptions.set_keep_alive(timeout);
+        if opts.mqtt_tls {
+            // Use rustls-native-certs to load root certificates from the operating system.
+            let mut root_cert_store = rustls::RootCertStore::empty();
+            for cert in
+                rustls_native_certs::load_native_certs().expect("could not load platform certs")
+            {
+                root_cert_store
+                    .add(&rustls::Certificate(cert.0))
+                    .expect("Failed to load DER from platform cert");
```

Comment:
should we perhaps add option to inject "pinned" cert pem's via metadata?

Reply:
I'm not sure how we can inject pinned certs at this moment. I think it'd involve a bunch of the stuff in the fluvio-sc to copy or mount stuff inside the container.

Reply:
The cert is just base64 string and it could be injected via metadata?